src/Stack/Solver.hs

```diff
@@ -560,8 +562,10 @@ cabalPackagesCheck cabalfps noPkgMsg dupErrMsg = do
     -- Just the latter check is enough to cover both the cases
 
     let packages  = zip cabalfps gpds
+        -- XXX taken from https://github.com/ppelleti/normalization-insensitive, see #1810
```

Comment:
Its just a String to Text conversion and back. I don't think you need this comment here.